cmd/acra-translator/acra-translator.go

```diff
@@ -191,7 +255,20 @@ func main() {
 		logging.SetLogLevel(logging.LogDiscard)
 	}
 
-	readerServer.Start(mainContext)
+	if os.Getenv(GracefulEnv) == "true" {
+		if *incomingConnectionGRPCString != "" && *incomingConnectionHTTPString != "" {
+			readerServer.StartFromFileDescriptor(mainContext, DescriptorHTTP, DescriptorGRPC)
+		} else {
+			if *incomingConnectionHTTPString != "" {
+				readerServer.StartFromFileDescriptor(mainContext, DescriptorHTTP, 0)
+			}
+			if *incomingConnectionGRPCString != "" {
+				readerServer.StartFromFileDescriptor(mainContext, 0, DescriptorGRPC)
+			}
+		}
```

Comment:
I don't quite understand what happens here. Could you please elaborate?

What I find strange is that zero is a valid file descriptor value (usually stdin), I don't think it's okay to use it like this.

Reply:
we should check it in function [StartFromFileDescriptor](https://github.com/cossacklabs/acra/blob/f8ab52028858926fd2ed645dd092a371cea62db5/cmd/acra-translator/server/server.go#L344). Add some value which will represent null as integer value

Reply:
Yes. That is weird piece of code that I wrote while debugging. Now I think we can do just `readerServer.StartFromFileDescriptor(mainContext, DescriptorHTTP, DescriptorGRPC)` in any case. Inside `StartFromFileDescriptor` we have `http`, `grpc` conditions that will use appropriate descriptors